bridges/snowbridge/pallets/inbound-queue-v2/src/lib.rs

```diff
@@ -239,11 +228,11 @@ pub mod pallet {
 
 			// Forward XCM to AH
 			let dest = Location::new(1, [Parachain(T::AssetHubParaId::get())]);
-			let message_id =
-				Self::send_xcm(dest.clone(), &relayer, xcm.clone()).map_err(|error| {
-					tracing::error!(target: LOG_TARGET, ?error, ?dest, ?xcm, "XCM send failed with error");
-					Error::<T>::from(error)
-				})?;
+
+			// Mark message as received
+			Nonce::<T>::set(nonce.into());
+
+			let send_result = Self::send_xcm(dest.clone(), &relayer, xcm.clone());
```

Comment:
Set the nonce even before the xcm is sent. I think this is better, with no chance for re-entrancy attacks.

Reply:
Agree, even though the whole transaction and storage changes would be reverted on error.

Reply:
I think that's the main argument for changing `send_xcm` to log an event rather than raise an error - it allows us to set the nonce. So perhaps the question is more, should the nonce increment if `send_xcm` fails? I think it should, since what happens after sending the xcm fails is out of our control, and could potentially have unexpected effects if the relayer can replay the message. Also, the likelihood of the message succeeding on a resubmission is low, so rather mark it as processed even though it failed.

Reply:
As I said in https://linear.app/snowfork/issue/SNO-1454#comment-5c6fa240, I'd like you to describe with some detail the attack vector. I can't see personally see any, which means we have a different understanding of the error handling.